packages/cli/src/utils/deploy.ts

```diff
@@ -397,13 +399,26 @@ export async function deploy(
     try {
       const factory = new ethers.ContractFactory(abi, bytecode, signer);
       console.log(chalk.gray(`executing deployment of ${contractName} with nonce ${nonce}`));
-      const deployPromise = factory
-        .deploy({
-          nonce: nonce++,
-          maxPriorityFeePerGas,
-          maxFeePerGas,
-        })
-        .then((c) => (disableTxWait ? c : c.deployed()));
+      let deployPromise;
+
+      if (maxFeePerGas && maxPriorityFeePerGas) {
+        deployPromise = factory
+          .deploy({
+            nonce: nonce++,
+            maxPriorityFeePerGas,
+            maxFeePerGas,
+          })
+          .then((c) => (disableTxWait ? c : c.deployed()));
+        /// Gas Price === Legacy
+      } else if (gasPrice) {
```

Comment:
similar to above we should check for `null` here, otherwise there might be issues with gas price 0